lms/validation/_base.py

```diff
@@ -94,6 +95,20 @@ def parse(self, *args, **kwargs):
     def _handle_error(error, _req, _schema, *, error_status_code, error_headers):
         raise ValidationError(messages=error.messages) from error
 
+    @marshmallow.pre_load
+    def _decode_jwt(self, data, **_kwargs):
+        """Use the values encoded in the `id_token` JWT if present."""
+        if "id_token" not in self.context["request"].POST:
+            return data
+
+        params = LTIParams.from_v13(self.context["request"].lti_jwt)
+        # Make the rest of params also accessible to marshmallow in case any are not coming from the JWT
+        # eg query parameters
+        # This is to make it backwards compatible with schemas that mix LTI parameters with others that belong to the LMS app (eg the `url` parameter).
+        params.update(self.context["request"].params)
```

Comment:
This extra parameters will be feed to the schema but ignored if not used, ie they won't appear to be present at all if not explicitly declared.